src/HttpApplicationRunner.php

```diff
@@ -81,9 +90,15 @@ public function __construct(
         string $vendorDirectory = 'vendor',
         string $configMergePlanFile = '.merge-plan.php',
         private readonly ?LoggerInterface $logger = null,
-        private readonly ?int $bufferSize = null,
+        ?int $bufferSize = null,
         private ?ErrorHandler $temporaryErrorHandler = null,
+        ?EmitterInterface $emitter = null,
+        private readonly bool $useRemoveBodyByStatusMiddleware = true,
+        private readonly bool $useContentLengthMiddleware = true,
+        private readonly bool $useHeadRequestMiddleware = true,
```

Comment:
Should we pass an array of middleware instead?

Reply:
No. These flags for backward compatibility only. In the next major they will be removed.